lib/spack/spack/cmd/buildcache.py

```diff
@@ -332,26 +337,25 @@ def _createtarball(env, spec_yaml=None, packages=None, add_spec=True,
                    signing_key=None, force=False, make_relative=False,
                    unsigned=False, allow_root=False, rebuild_index=False):
     if spec_yaml:
-        packages = set()
         with open(spec_yaml, 'r') as fd:
             yaml_text = fd.read()
             tty.debug('createtarball read spec yaml:')
             tty.debug(yaml_text)
-            s = Spec.from_yaml(yaml_text)
-            packages.add('/{0}'.format(s.dag_hash()))
+            matches = find_matching_specs([Spec.from_yaml(yaml_text)], env=env)
```

Comment:
This appears to be the only one of the three uses of `find_matching_specs` that does not call `spack.cmd.parse_specs()`

Reply:
Let me just restore that to what it was as well, the important part is that `elif env:` does not call `find_matching_specs` anymore